source/DistanceMap.cpp

```diff
@@ -175,19 +150,23 @@ bool DistanceMap::Edge::operator<(const Edge &other) const
 // source system or the maximum count is reached.
 void DistanceMap::Init(const Ship *ship)
 {
-	if(!center || (ship && ship->IsRestrictedFrom(*center)))
+	if(!center || (ship && ship->IsRestrictedFrom(*center)) || center == destination)
 		return;
 
-	route[center] = Edge();
-	if(!maxDistance)
+	// To get to the starting point, there is no previous system,
+	// and it takes no fuel or days.
+	route[center] = RouteEdge();
+	if(!maxDays)
 		return;
 
-	// Check what travel capabilities this ship has. If no ship is given, assume
-	// hyperdrive capability and no jump drive.
+	// Check what travel capabilities this ship has. If no ship is given, the
+	// DistanceMap class defaults assume hyperdrive capability only.
 	if(ship)
 	{
 		this->ship = ship;
 		hyperspaceFuel = ship->JumpNavigation().HyperdriveFuel();
+		// Todo: consider outfit "jump distance" at each link to find if more fuel is needed
+		// by a second jump drive outfit with more range and cost via JumpDriveFuel(to).
```

Comment:
Is this something that our current implementation can do, or just a new feature that would be nice to have?

Reply:
It seems the engine handles it with jumpDriveCosts (even though it's not used at all?), but certainly the DistanceMap would not be accurate since it uses the maximum jump range between multiple jump drives, but minimum jump cost, sometimes overestimating how far it could get with a given amount of fuel.

Reply:
And I say "it's not used" meaning there's just no content in the game. As far as I can see, GetCheapestJumpType means the correct jump drive would be used if there were multiple jump drives with different range/costs.